Gems/PhysX/Code/Tests/EditorColliderComponentTests.cpp

```diff
@@ -6,138 +6,609 @@
+        // the collider component should be compatible with multiple collider components
+        entity->CreateComponent<PhysX::EditorMeshColliderComponent>();
+        entity->CreateComponent<PhysX::EditorShapeColliderComponent>();
+        entity->CreateComponent(LmbrCentral::EditorBoxShapeComponentTypeId);
 
-         m_actionDispatcher->CameraState(m_cameraState);
+        // the entity should be in a valid state because the component requirement is satisfied
+        AZ::Entity::DependencySortOutcome sortOutcome = entity->EvaluateDependenciesGetDetails();
+        EXPECT_TRUE(sortOutcome.IsSuccess());
     }
 
-    AzToolsFramework::EntityIdList ColliderPickingFixture::ClickAndGetSelectedEntities(AzFramework::ScreenPoint screenPoint)
+    TEST_F(PhysXEditorFixture, EditorColliderComponent_ColliderWithBox_CorrectRuntimeComponents)
     {
-        // click the entity in the viewport
-        m_actionDispatcher->SetStickySelect(true)
-            ->CameraState(m_cameraState)
-            ->MousePosition(screenPoint)
-            ->MouseLButtonDown()
-            ->MouseLButtonUp();
+        // create an editor entity with a collider component
+        EntityPtr editorEntity = CreateInactiveEditorEntity("ColliderComponentEditorEntity");
+        auto* colliderComponent = editorEntity->CreateComponent<PhysX::EditorColliderComponent>();
+        editorEntity->CreateComponent<PhysX::EditorStaticRigidBodyComponent>();
+        editorEntity->Activate();
+
+        // Set collider to be a box
+        AZ::EntityComponentIdPair idPair(editorEntity->GetId(), colliderComponent->GetId());
+        PhysX::EditorPrimitiveColliderComponentRequestBus::Event(idPair, &PhysX::EditorPrimitiveColliderComponentRequests::SetShapeType, Physics::ShapeType::Box);
+
+        EntityPtr gameEntity = CreateActiveGameEntityFromEditorEntity(editorEntity.get());
 
-        AzToolsFramework::EntityIdList selectedEntities;
-        AzToolsFramework::ToolsApplicationRequestBus::BroadcastResult(
-            selectedEntities, &AzToolsFramework::ToolsApplicationRequestBus::Events::GetSelectedEntities);
-        return selectedEntities;
+        // check that the runtime entity has the expected components
+        EXPECT_TRUE(gameEntity->FindComponent<PhysX::BoxColliderComponent>() != nullptr);
+        EXPECT_TRUE(gameEntity->FindComponent<PhysX::StaticRigidBodyComponent>() != nullptr);
     }
 
-    TEST_F(ColliderPickingFixture, ColliderPickingWithBoxShape)
+    TEST_F(PhysXEditorFixture, EditorColliderComponent_ColliderWithBox_CorrectRuntimeGeometry)
     {
-         // Given the setup conditions
-         const AZ::Vector3 boxDimensions(5.0f, 5.0f, 5.0f);
-         SetupCollider(Physics::BoxShapeConfiguration(boxDimensions), ShapeRotation, ShapeOffset);
-         SetupTransform(EntityRotation, EntityTranslation, uniformScale);
+        // create an editor entity with a collider component
```

Comment:
Yes, I'll change it for the other types too.

Reply:
On a second thought, I'll leave it like that for the first function because it doesn't set properties and it's a good way of testing default values, the helper function always call the set properties functions.